File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        String answer = "";
        answer = new_id.toLowerCase();
        answer = answer.replaceAll("[^\\w\\-_.]*", "");
        answer = answer.replaceAll("\\.{2,}", ".");
        answer = answer.replaceAll("^[.]|[.]$", "");
        answer = answer.length() == 0 ? "a" : answer;
        
        if(answer.length() >= 16) answer = answer.substring(0, 15).replaceAll("^[.]|[.]$", "");
        
        if(answer.length() <= 2) {
            char c = answer.charAt(answer.length() - 1);
            
            for(int i = 0; i<=3-answer.length(); i++){
                answer += c;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    var answer = '';
    answer = new_id.toLowerCase().match(/[a-z0-9-_.]/g).join('').replace(/\.+/g, ".")

    if(answer.startsWith(".")) answer = answer.substring(1, answer.length)
    else if(answer.endsWith(".")) answer = answer.substring(0, answer.length - 1)
    
    if(answer.length === 0) answer = "a"
    else if(answer.length >= 16) answer = answer.substr(0, 15)
    
    if(answer.endsWith(".")) answer = answer.substring(0, answer.length - 1)
    
    while(answer.length <= 2) answer += answer[answer.length - 1]
    
    return answer;
}